src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
+
+    response.status(status).json(APIResponse.create(0, status, error));
```

Comment:
PullProofAI

The logic for constructing the `error` variable can be simplified and made more readable by refactoring it into a separate function.
```diff
+   private getErrorMessage(exception: HttpException): string {
+       const response = exception.getResponse();
+       if (typeof response === 'string') {
+           return response;
+       } else if (response && typeof response['message'] === 'object') {
+           return response['message'].join(', ');
+       } else {
+           return response['message'];
+       }
+   }
+
+   catch(exception: HttpException, host: ArgumentsHost) {
+       let err = exception['response'];
+       err = err.response ? err.response : JSON.stringify(err);
+       const context = host.switchToHttp();
+       const response = context.getResponse<Response>();
+       const status = exception.getStatus();
+       this.logger.error(
+           `Message : ${err}`,
+           `${context.getRequest().method} ${context.getRequest().url}`,
+       );
+       const error = this.getErrorMessage(exception);
+
+       response.status(status).json(APIResponse.create(0, status, error));
+   }
```



<!-- This is an auto-generated comment by PullProofAI -->